packages/slate-editor/src/components/EditorBlock/ResizableEditorBlock.tsx

```diff
@@ -98,53 +95,59 @@ export const ResizableEditorBlock = forwardRef<HTMLDivElement, Props>((props, re
         [width, containerWidth, constrainSize],
     );
 
-    function renderFrame(props: { isSelected: boolean }) {
+    function renderFrame(props: { isSelected: boolean; isHovered: boolean }) {
         return (
             <>
                 {renderInjectionPoint(renderReadOnlyFrame ?? renderEditableFrame, props)}
-                {resizable && props.isSelected && (
-                    <Draggable.DraggableCore
-                        offsetParent={blockElement ?? undefined}
-                        onDrag={handleResizeEvent}
-                        onStart={handleResizingStarted}
-                        onStop={handleResizingFinished}
-                    >
-                        <div contentEditable={false}>
-                            <ResizeButton
-                                className={classNames(
-                                    styles.ResizeButton,
-                                    isInvertedResizing ? styles.left : styles.right,
-                                )}
-                                position={isInvertedResizing ? 'left' : 'right'}
-                            />
-                        </div>
-                    </Draggable.DraggableCore>
+                {resizable && (props.isSelected || props.isHovered) && (
+                    <div contentEditable={false}>
+                        <ResizeButton
+                            offsetParent={blockElement ?? undefined}
+                            onDrag={handleResizeEvent}
+                            onStart={handleResizingStarted}
+                            onStop={handleResizingFinished}
+                            className={classNames(styles.ResizeButton, styles.left)}
+                            position={'left'}
+                        />
+                        <ResizeButton
+                            offsetParent={blockElement ?? undefined}
+                            onDrag={handleResizeEvent}
+                            onStart={handleResizingStarted}
+                            onStop={handleResizingFinished}
+                            className={classNames(styles.ResizeButton, styles.right)}
+                            position={'right'}
+                        />
+                    </div>
```

Comment:
As a big fan of transitions :sweat_smile: , I suggest adding a small transition from `opacity: 0` to `opacity: 1` for the resize  handles. It's much more pleasant. 

```diff
Index: packages/slate-editor/src/components/EditorBlock/ResizableEditorBlock.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/slate-editor/src/components/EditorBlock/ResizableEditorBlock.tsx b/packages/slate-editor/src/components/EditorBlock/ResizableEditorBlock.tsx
--- a/packages/slate-editor/src/components/EditorBlock/ResizableEditorBlock.tsx	(revision 4388c36bc98786430b7c0f6240ead15bc67b3d84)
+++ b/packages/slate-editor/src/components/EditorBlock/ResizableEditorBlock.tsx	(date 1666692569794)
@@ -99,7 +99,7 @@
         return (
             <>
                 {renderInjectionPoint(renderReadOnlyFrame ?? renderEditableFrame, props)}
-                {resizable && (props.isSelected || props.isHovered) && (
+                {resizable && (
                     <div contentEditable={false}>
                         <ResizeButton
                             offsetParent={blockElement ?? undefined}
@@ -107,7 +107,8 @@
                             onStart={handleResizingStarted}
                             onStop={handleResizingFinished}
                             className={classNames(styles.ResizeButton, styles.left)}
-                            position={'left'}
+                            position="left"
+                            show={props.isSelected || props.isHovered}
                         />
                         <ResizeButton
                             offsetParent={blockElement ?? undefined}
@@ -115,7 +116,8 @@
                             onStart={handleResizingStarted}
                             onStop={handleResizingFinished}
                             className={classNames(styles.ResizeButton, styles.right)}
-                            position={'right'}
+                            position="right"
+                            show={props.isSelected || props.isHovered}
                         />
                     </div>
                 )}
Index: packages/slate-editor/src/components/EditorBlock/ResizeButton.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/slate-editor/src/components/EditorBlock/ResizeButton.tsx b/packages/slate-editor/src/components/EditorBlock/ResizeButton.tsx
--- a/packages/slate-editor/src/components/EditorBlock/ResizeButton.tsx	(revision 4388c36bc98786430b7c0f6240ead15bc67b3d84)
+++ b/packages/slate-editor/src/components/EditorBlock/ResizeButton.tsx	(date 1666692249289)
@@ -15,6 +15,7 @@
     onStop: (e: DraggableEvent, data: DraggableData, position: Position) => void | false;
     className?: string;
     offsetParent?: Draggable.DraggableCoreProps['offsetParent'];
+    show: boolean;
 }
 
 export function ResizeButton({
@@ -24,6 +25,7 @@
     onStart,
     onDrag,
     onStop,
+    show,
 }: Props) {
     return (
         <Draggable.DraggableCore
@@ -32,7 +34,7 @@
             onDrag={(event, data) => onDrag(event, data, position)}
             onStop={(event, data) => onStop(event, data, position)}
         >
-            <div className={classNames(styles.handle, className)}>
+            <div className={classNames(className, styles.handle, { [styles.show]: show })}>
                 <button className={styles.button} onMouseDown={preventDefault} type="button" />
             </div>
         </Draggable.DraggableCore>
Index: packages/slate-editor/src/components/EditorBlock/ResizeButton.module.scss
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/slate-editor/src/components/EditorBlock/ResizeButton.module.scss b/packages/slate-editor/src/components/EditorBlock/ResizeButton.module.scss
--- a/packages/slate-editor/src/components/EditorBlock/ResizeButton.module.scss	(revision 4388c36bc98786430b7c0f6240ead15bc67b3d84)
+++ b/packages/slate-editor/src/components/EditorBlock/ResizeButton.module.scss	(date 1666692450522)
@@ -15,4 +15,13 @@
     border: 1px solid rgba(255, 255, 255, 0.9);
     width: 6px;
     height: 48px;
+
+    pointer-events: none;
+    opacity: 0;
+    transition: opacity 100ms linear;
+
+    .show & {
+        pointer-events: all;
+        opacity: 1;
+    }
 }
```